realsense_ros_camera/src/realsense_node_factory.cpp

```diff
@@ -79,6 +79,49 @@ void RealSenseNodeFactory::onInit()
             exit(1);
         }
 
+        // we found the device. Restart the device and wait for it to come up again
+
+        ROS_INFO_STREAM("RESETING DEVICE: " << usb_port_id);
+        _device.hardware_reset();
+        ros::Duration(5).sleep();
+        ROS_INFO_STREAM("Attempting to reacquire");
+
+        list = _ctx.query_devices();
+        if (0 == list.size())
+        {
+            ROS_ERROR("No RealSense devices were found! Terminating RealSense Node...");
+            ros::shutdown();
+            exit(1);
+        }
+        found = false;
+        for (auto&& dev : list)
+        {
+            auto sn = dev.get_info(RS2_CAMERA_INFO_SERIAL_NUMBER);
+            auto port_id = parseUsbPortId(dev.get_info(RS2_CAMERA_INFO_PHYSICAL_PORT));
+            if (usb_port_id.empty())
```

Comment:
This loop is not guaranteed to select the same device the previous one did (the one that was just reset). So, the returned device may not, in fact, have been reset properly.